lucien-ios/AddAndEditNewComicBook/ComicFormViewController.swift

```diff
@@ -74,6 +85,27 @@ final class AddComicViewController: UIViewController, AlertDisplaying {
         showAlert(title: "", message: "This will delete your current comic information.", actions: [goBackAction, cancelAction], preferredStyle: .actionSheet)
     }
 
+    private func fillFieldsWithComicFormViewModelData() {
+        guard let comicFormViewModel = comicFormViewModel else { return }
+        if comicFormViewModel.comicFormMode == .edit {
+            seriesTitleTextField.text = comicFormViewModel.seriesTitle ?? ""
+            volumeTextField.text = comicFormViewModel.volume ?? ""
+            storyTitleTextField.text = comicFormViewModel.storyTitle
+            issueTextField.text = comicFormViewModel.issue ?? ""
+            publisherTextField.text = comicFormViewModel.publisher ?? ""
+            releaseDateTextField.text = comicFormViewModel.release ?? ""
+            genrePicker.isHidden = false
+            conditionPicker.isHidden = false
+            selectAGenreButton.setTitle(comicFormViewModel.genre?.title, for: .normal)
+            selectAConditionButton.setTitle(comicFormViewModel.condition?.title, for: .normal)
+            genrePicker.selectRow(comicFormViewModel.genre?.rawValue ?? 0, inComponent: 0, animated: false)
+            conditionPicker.selectRow(comicFormViewModel.condition?.rawValue ?? 0, inComponent: 0, animated: false)
+            if let coverPhoto = comicFormViewModel.coverPhoto {
+                self.cameraViewController(didCapture: coverPhoto)
```

Comment:
It's never a good idea to directly call a delegate method. Instead, extract the body of the delegate method into another function (called `updateCoverPhotoButton` or similar) and call that here. Rationale: eventually you may want to add side effects other than just the update to the delegate method. You probably don't want those side effects here.